SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialEventTracer.cpp

```diff
@@ -20,37 +20,36 @@ void SpatialEventTracer::TraceCallback(void* UserData, const Trace_Item* Item)
 		return;
 	}
 
-	uint32_t ItemSize = Trace_GetSerializedItemSize(Item);
 	// Depends whether we are using rotating logs or single-log mode (where we track max size).
 	const bool bTrackFileSize = EventTracer->MaxFileSize != 0;
-	if (!bTrackFileSize || (EventTracer->BytesWrittenToStream + ItemSize <= EventTracer->MaxFileSize))
+	if (!bTrackFileSize || (EventTracer->BytesWrittenToStream < EventTracer->MaxFileSize))
 	{
-		if (bTrackFileSize)
-		{
-			EventTracer->BytesWrittenToStream += ItemSize;
-		}
+		int64_t BytesWritten = Trace_SerializeItemToStream(Stream, Item);
 
-		int Code = Trace_SerializeItemToStream(Stream, Item, ItemSize);
-		if (Code == WORKER_RESULT_FAILURE)
+		if (BytesWritten == WORKER_RESULT_FAILURE)
 		{
-			UE_LOG(LogSpatialEventTracer, Error, TEXT("Failed to serialize to with error code %d (%s)"), Code,
+			UE_LOG(LogSpatialEventTracer, Error, TEXT("Failed to serialize with error code %lld (%s)"), BytesWritten,
 				   ANSI_TO_TCHAR(Trace_GetLastError()));
 		}
+		else if (BytesWritten == 0)
+		{
+			UE_LOG(LogSpatialEventTracer, Error,
+				   TEXT("Failed to serialize due to file stream having insufficient capacity. This shouldn't happen with file streams."));
```

Comment:
Why should this not happen?

Reply:
from the new 15.2.0 docstring for Trace_SerializeItemToStream
```
 * Returns 0 if the stream had insufficient capacity to write the serialized item, and nothing was
 * written. This should never happen for a stream with conceptually infinite capacity (e.g a file
 * stream). For streams with finite capacity, reading from the stream may make room.
 ```
 Depending on worker implementation, perhaps this could happen if a user runs out of space on their filesystem, I'll ammend the error message slightly to simply `Failed to serialize due to file stream having insufficient capacity.`

Reply:
Cooool I think I was just trying to figure in my head what happens with the two different type of stream we use: "rotating log" and a standard stream. Think this makes sense!